src/GrpcProtos/capture.proto

```diff
@@ -599,6 +599,30 @@ message ThreadStateSlice {
   // Otherwise, they should be 0 to minimize bandwidth usage.
   uint32 wakeup_tid = 7;
   uint32 wakeup_pid = 8;
+
+  // The callstacks for the tracepoints get split during processing and
+  // originate from a different ProducerCaptureEvent. So the callstacks are not
+  // part of the original ProducerCaptureEvent, but will be added by the
+  // ProducerEventProcessor.
+  // If callstack collection is disabled, `kNoCallstack` expresses that there
+  // is no callstack to wait for.
+  // On `kWaitingForCallstack`, the ProducerEventProcessor knows to wait with
+  // sending this event until the callstack has been received.
+  // Once added a callstack, `kCallstackSet` expresses to the client that there
+  // is a valid callstack id set.
+  enum CallstackStatus {
+    kNoCallstack = 0;
+    kWaitingForCallstack = 1;
+    kCallstackSet = 2;
+  }
+  CallstackStatus switch_out_or_wakeup_callstack_status = 10;
+
+  // This is the callstack of the tracepoint that changed the state
+  // of the thread and caused this slice to happen. The callstack corresponds to
+  // the beginning of the slice.
+  // If the thread state is runnable then this callstack is coming from
```

Comment:
Carrying this over from the previous review round:

We have wakeup_reason that controls whether wakeup_tid is valid, but then we have the state itself that controls whether triggering_callstack_id is from this thread or wakeup_tid. This is explained here but is still awkward when using the proto.

Would it correct to say that this callstack comes from this thread if wakeup_reason is kNotApplicable, and from wakeup_tid otherwise?

Reply:
For this round of reviews. Yep, I think it is correct.

Reply:
Then I think we should write it in those terms for the reasons I mentioned: it's better to reason entirely in terms of the `wakeup_` fields rather than having to involve the specific state (and keep track which state implies one thing and which implies the other).

Reply:
Agree. Better now?

Reply:
Yep, thanks.